azurelinuxagent/ga/update.py

```diff
@@ -351,8 +342,20 @@ def run(self, debug=False):
                     py_micro=PY_VERSION_MICRO, systemd=systemd.is_systemd(),
                     lis_ver=get_lis_version(), has_logrotate=has_logrotate()
                 )
-
             logger.info(os_info_msg)
+
+            #
+            # Initialize the goal state; some components depend on information provided by the goal state and this
+            # call ensures the required info is initialized (e.g. telemetry depends on the container ID.)
+            #
+            protocol = self.protocol_util.get_protocol()
+
+            self._initialize_goal_state(protocol)
+
+            # Initialize the common parameters for telemetry events
+            initialize_event_logger_vminfo_common_parameters(protocol)
+
+            # Send telemetry for the OS-specific info.
             add_event(AGENT_NAME, op=WALAEventOperation.OSInfo, message=os_info_msg)
```

Comment:
is this intentional? why don't we keep this line next to the logger.

Reply:
telemetry depends on the goal state, so we need to initialize the gs first (line 353)

Reply:
I don't think it's too big of a deal, but is there any reason not to move the `logger.info` line down till after grabbing the goal state?

Reply:
yes, the log messages wrt the goal state tend to obscure the ID info. The intention is to have all the information that identifies the agent at the top of the log